components/dashboard/application/environment/show.tsx

```diff
@@ -1,42 +1,30 @@
-import { CodeEditor } from "@/components/shared/code-editor";
-import { Button } from "@/components/ui/button";
-import {
-	Card,
-	CardContent,
-	CardDescription,
-	CardHeader,
-	CardTitle,
-} from "@/components/ui/card";
-import {
-	Form,
-	FormControl,
-	FormField,
-	FormItem,
-	FormMessage,
-} from "@/components/ui/form";
-import { Toggle } from "@/components/ui/toggle";
+import { Form } from "@/components/ui/form";
+import { Secrets } from "@/components/ui/secrets";
 import { api } from "@/utils/api";
 import { zodResolver } from "@hookform/resolvers/zod";
-import { EyeIcon, EyeOffIcon } from "lucide-react";
-import React, { useEffect, useState } from "react";
 import { useForm } from "react-hook-form";
 import { toast } from "sonner";
 import { z } from "zod";
 
 const addEnvironmentSchema = z.object({
-	environment: z.string(),
+	env: z.string(),
 });
 
 type EnvironmentSchema = z.infer<typeof addEnvironmentSchema>;
 
+const addBuildArgsSchema = z.object({
+	buildArgs: z.string(),
+});
+
+type BuildArgsSchema = z.infer<typeof addBuildArgsSchema>;
+
 interface Props {
 	applicationId: string;
 }
 
 export const ShowEnvironment = ({ applicationId }: Props) => {
-	const [isEnvVisible, setIsEnvVisible] = useState(true);
-	const { mutateAsync, isLoading } =
-		api.application.saveEnvironment.useMutation();
+	const saveEnvironmentMutation = api.application.saveEnvironment.useMutation();
+	const saveBuildArgsMutation = api.application.saveBuildArgs.useMutation();
```

Comment:
Instead of multiple routers, use `update` so we can use the same endpoint for both operations

Reply:
Updated keeping one operation!